### PR TITLE
Add nginx as sidecar to staging deployment

### PR DESCRIPTION
**Pull request by sonalkr132**
We are hoping to migrate away from existing nginx LB, which exists outside of kubernetes cluster.
nginx config is based on [rubygems-infrastructure/rubygems-balancer](https://github.com/rubygems/rubygems-infrastructure/blob/af8694a1efff3cd9d05a60d07486f5669fb9dd1c/cookbooks/rubygems-balancer/templates/default/site.conf.erb). We are not adding ssl config
as we intend to offload it on kubernetes ingress.

### CODE REVIEW

**Comment by sonalkr132** [comment]
One notable difference from the chef module is that I haven't set [ulimit](https://github.com/rubygems/rubygems-infrastructure/blob/af8694a1efff3cd9d05a60d07486f5669fb9dd1c/cookbooks/rubygems-balancer/files/default/nginx#L1). I doubt it will be needed when nginx is running on each pod. Haven't added datadog agent, I am hoping existing logging pipeline should work.
Also, [maintenance.html](https://github.com/rubygems/rubygems-infrastructure/blob/af8694a1efff3cd9d05a60d07486f5669fb9dd1c/cookbooks/rubygems-balancer/files/default/maintenance.html) hasn't been added. configmap should be enough or do we want add it with a Dockerfile for nginx?

File: config/deploy/staging/nginx-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nginx-configmap
+  namespace: rubygems-staging
```
Comment (sonalkr132) [inline config/deploy/staging/nginx-configmap.yaml:5]: does this namespace need to be removed? I think kubernetes-deploy adds it for the rest of the resources.

Comment (dwradcliffe) [reply config/deploy/staging/nginx-configmap.yaml:5]: yeah this is unnecessary since doing the `apply will add it automatically.

File: config/environments/staging.rb [hunk 2]
```diff
@@ -51,13 +51,7 @@
   # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  config.force_ssl = true
-  config.ssl_options = {
-    hsts: { expires: 2.minutes, subdomains: false },
-    redirect: {
-      exclude: ->(request) { request.path.start_with?('/internal') }
-    }
-  }
```
Comment (dwradcliffe) [inline config/environments/staging.rb:60]: Why did you remove this?

Comment (sonalkr132) [reply config/environments/staging.rb]: I didn't realize at the time this works with `X-Forwarded-Proto`. I assumed this will cause redirect loop if we offload on ALB. Good to revert :+1:

File: config/deploy/staging/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            backend:
+              serviceName: unicorn
+              servicePort: 80
+    - host: staging-origin.rubygems.org
+      http:
+        paths:
+          - path: /*
+            backend:
+              serviceName: unicorn
+              servicePort: 80
```
Comment (sonalkr132) [inline config/deploy/staging/ingress.yaml:24]: Doesn't `alb.ingress.kubernetes.io/target-type: instance` need `NodePort` service?

Comment (sonalkr132) [reply config/deploy/staging/ingress.yaml:24]: We could also add `alb.ingress.kubernetes.io/healthcheck-path: /internal/ping`

Comment (dwradcliffe) [reply config/deploy/staging/ingress.yaml:28]: LoadBalancer is also a NodePort service. But I changed this to target-type: ip and set the healthcheck path like you suggested.
